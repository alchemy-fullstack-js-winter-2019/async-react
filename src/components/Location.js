import React from 'react';
import propTypes from 'prop-types';

function Location({ character }) {
  const { name, dimension }
}