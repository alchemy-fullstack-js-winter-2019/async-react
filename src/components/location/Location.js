import React from 'react';
import PropTypes from 'prop-types';

function Location({ location }) {
  const { name, dimension, type } = location;

  return (
    <>
    <h2> {name} </h2>
    <h2> {dimension} </h2>
    <h2> {type} </h2>
    </>
  );
}

Location.propTypes = {
  location: PropTypes.object.isRequired
};

export default Location;
