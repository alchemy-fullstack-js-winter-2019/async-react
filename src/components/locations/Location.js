import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Location extends PureComponent {
  static propTypes = {
    result: PropTypes.object.isRequired
  }

  render() {
    const { name, type, dimension } = this.props.result;

    return (
      <>
        <p>Location: {name}</p>
        <p>Type: {type}</p>
        <p>Dimension: {dimension}</p>
      </>
    );
  }
}
