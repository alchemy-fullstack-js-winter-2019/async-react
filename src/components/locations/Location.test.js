import React from 'React';
import renderer from 'react-test-renderer';
import Location from './Location';

describe('Location', () => {
  it('matches a snapshot', () => {
    const location = {
      name: 'Earth (C-137)',
      type: 'Planet',
      dimension: 'Dimension C-137'
    };
    const tree = renderer.create(<Location location={location} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
