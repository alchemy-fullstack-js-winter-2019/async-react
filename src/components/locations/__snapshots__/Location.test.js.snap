// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location matches a snapshot 1`] = `
Array [
  <h2>
    Earth (C-137)
  </h2>,
  <p>
    Planet
     | 
    Dimension C-137
  </p>,
]
`;
